src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -0,0 +1,778 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    /// <summary>
+    /// Represents a C# or VB expression.
+    /// </summary>
+    public interface IExpression : IOperation
+    {
+        /// <summary>
+        /// Result type of the expression.
+        /// </summary>
+        ITypeSymbol ResultType { get; }
+        /// <summary>
+        /// If the expression evaluates to a constant value, the value of the expression, and otherwise null.
+        /// </summary>
+        object ConstantValue { get; }
+    }
+
+    /// <summary>
+    /// Represents a C# or VB method invocation.
+    /// </summary>
+    public interface IInvocation : IExpression
+    {
+        /// <summary>
+        /// Method to be invoked.
+        /// </summary>
+        IMethodSymbol TargetMethod { get; }
+        /// <summary>
+        /// 'This' or 'Me' argument to be supplied to the method.
+        /// </summary>
+        IExpression Instance { get; }
+        /// <summary>
+        /// Kind of the invocation.
+        /// </summary>
+        InvocationKind InvocationKind { get; }
+        /// <summary>
+        /// Arguments of the invocation, excluding the instance argument.
+        /// </summary>
+        ImmutableArray<IArgument> Arguments { get; }
+        /// <summary>
+        /// Find the argument supplied for a given parameter of the target method.
+        /// </summary>
+        /// <param name="parameter">Parameter of the target method.</param>
+        /// <returns>Argument corresponding to the parameter.</returns>
+        IArgument ArgumentMatchingParameter(IParameterSymbol parameter);
```

Comment:
If it's worth having it should be on the interface, because an extension method would need to compute all the arguments in order to find the one of interest.